ledger_test.go

```diff
@@ -0,0 +1,115 @@
+// +build !speculos
+
+package ledger
+
+import (
+	"fmt"
+	"testing"
+)
+
+// Return string representation of transaction type
+func getTxTypeString(txType byte) string {
+	switch txType {
+	case 0:
+		return "COIN ED"
+	case 2:
+		return "EXEC APP ED"
+	case 4:
+		return "SPAWN APP ED"
+	default:
+		return "UNWNOWN"
+	}
+}
+
+// Display transaction info
+func printTxInfo(txInfo *txInfo) {
+	fmt.Printf("Check tx params on ledger:\n")
+	fmt.Printf("\tTx type: %s\n", getTxTypeString(txInfo.Type))
+	fmt.Printf("\tSend SMH: %v\n", float64(txInfo.Amount)/1000000000000.0)
+	fmt.Printf("\tTo address: %x\n", txInfo.To)
+	fmt.Printf("\tMax Tx Fee: %v\n", float64(txInfo.GasLimit*txInfo.GasPrice)/1000000000000.0)
+	fmt.Printf("\tSigner: %x\n", txInfo.PublicKey[:20])
+}
+
+// Run tests on real Ledger device
+func doLedgerTest(t *testing.T, device *Ledger) bool {
+	// open Ledger device
+	if err := device.Open(); err != nil {
+		fmt.Printf("Open Ledger ERROR: %v\n", err)
+		return false
+	}
+	defer device.Close()
+
+	// run GetVersion test
+	fmt.Printf("GetVersion test:\n")
+	version, err := device.GetVersion()
+	if err != nil {
+		fmt.Printf("get version ERROR: %v\n", err)
+		return false
+	}
+	fmt.Printf("OK, Version: %+v\n", version)
+
+	// run GetExtendedPublicKey test
+	fmt.Printf("GetExtendedPublicKey test: Follow Ledger display\n")
+	publicKey, err := device.GetExtendedPublicKey(StringToPath("44'/540'/0'/0/0'"))
+	if err != nil {
+		fmt.Printf("get public key ERROR: %v\n", err)
+		return false
+	}
+	fmt.Printf("OK, public key: %x\n", publicKey)
+
+	// run GetAddress test
+	fmt.Printf("GetAddress test: Follow Ledger display\n")
+	address, err := device.GetAddress(StringToPath("44'/540'/0'/0/0'"))
+	if err != nil {
+		fmt.Printf("get address ERROR: %v\n", err)
+		return false
+	}
+	fmt.Printf("OK, address: %x\n", address)
+
+	// run ShowAddress test
+	fmt.Printf("ShowAddress test: Follow Ledger display\n")
```

Comment:
```suggestion
	fmt.Printf("ShowAddress test: Please make sure the following address agrees with your Ledger display.\n")
```

(print the path here, too)